File: core/src/kr/edoli/imview/ui/drawable/ColorDrawable.java
package kr.edoli.imview.ui.drawable;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.scenes.scene2d.utils.BaseDrawable;
import kr.edoli.imview.ui.res.Textures;

/**
 * Created by 석준 on 2016-02-06.
 */
public class ColorDrawable extends BaseDrawable {

    private Color color;

    public ColorDrawable(Color color) {
        this.color = color;
    }

    @Override
    public void draw(Batch batch, float x, float y, float width, float height) {
        Color preColor = batch.getColor();

        batch.setColor(color);
        batch.draw(Textures.White, x, y, width, height);

        batch.setColor(preColor);

        super.draw(batch, x, y, width, height);
    }
}

File: core/src/kr/edoli/imview/ui/ColorDrawable.kt
package kr.edoli.edoliui.widget.drawable

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.scenes.scene2d.utils.BaseDrawable
import kr.edoli.imview.ui.drawRect
import kr.edoli.imview.ui.drawRoundRect

/**
 * Created by daniel on 16. 6. 5.
 */
class ColorDrawable(
        private val color: Color,
        private val round: Float = 0f) : BaseDrawable() {

    override fun draw(batch: Batch, x: Float, y: Float, width: Float, height: Float) {
        val preColor = batch.color

        batch.color = color

        if (round == 0f) {
            batch.drawRect(x, y, width, height)
        } else {
            batch.drawRoundRect(x, y, width, height, round)
        }

        batch.color = preColor

        super.draw(batch, x, y, width, height)
    }
}
